### PR TITLE
Feature/85 collection form

### LINKED ISSUE #85: Add Collection form
Add a form for MU employees to create a Collection
- Include text inputs for `title` and `description`
- Reuse the mural search bar from [Search.tsx](https://github.com/hack4impact-mcgill/mu-map/blob/main/frontend/src/Search/Search.tsx)
- Use a multi-add like in [muralForm.tsx](https://github.com/hack4impact-mcgill/mu-map/blob/main/frontend/src/muralForm/MuralForm.tsx#L102) for `muralsInCollection` 
- Use searchCard.tsx to render the added murals
- Use the `POST` `/collection` endpoint (first try it in Postman) to create a new Collection

See [Figma](https://www.figma.com/file/1C71g93ZA1yFWchMa4id7J/MU-Map-2020-21?node-id=639%3A688) for layout:

Screen Shot 2021-01-27 at 9 36 24 PM

Screen Shot 2021-01-27 at 9 40 08 PM

### PR DESCRIPTION
**Pull request by tedspare**
# Summary

Add collection form based on the new Figma:

- Add fields for `title`, `description`, and a `mural` search bar
- Since fields aren't outlined, show a coloured pencil icon when the user hovers over an editable field
- Loop over and render @yi-fang-yuan 's `searchCard`s with the added murals
- Show a success badge when a collection is successfully `POST`ed

Some other fun stuff:

- Add an `ActionButtons` component in the process, which takes a `cancel` and `save` callback
- Adjust sidebar CSS to match new Figma

To do:

- Handle cancel button event to close sidebar
- Parametrize field values as props to allow a pin to be loaded into the form
- Parametrize whether the fields can be edited

## Test Plan

Some screenshots:
Screen Shot 2021-02-01 at 3 55 42 PM

Screen Shot 2021-02-01 at 3 55 42 PM

## Related Issues

Closes #85

### CODE REVIEW

File: frontend/src/App/App.tsx [hunk 1]
```diff
@@ -65,7 +66,7 @@ function App() {
   };
 
   const getMural = async () => {
-    const response = await fetch("http://localhost:3000/mural");
+    const response = await fetch("http://mumapbackend.us-east-1.elasticbeanstalk.com/mural");
```
Comment (JDziewonski98) [inline frontend/src/App/App.tsx:69]: better to not leave this hardcoded

File: frontend/src/CollectionForm/CollectionForm.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  const [murals, setMurals] = useState<any>([]);
+  const [muralsInCollection, setMuralsInCollection] = useState<any>([]);
+  const [muralResults, setMuralResults] = useState<any>([]);
+  const [muralQuery, setMuralQuery] = useState<string>("");
+
+  const [description, setDescription] = useState([]);
+  const [editingDesc, setEditingDesc] = useState<boolean>(false);
+  const [hoveringDesc, setHoveringDesc] = useState<boolean>(false);
+
+  const [title, setTitle] = useState([]);
```
Comment (JDziewonski98) [inline frontend/src/CollectionForm/CollectionForm.tsx:48]: nitpick: add <any> to stay consistent

File: frontend/src/CollectionForm/CollectionForm.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    if (muralsInCollection.some((currMural: any) => currMural.id === addedMural.id))
+      return
+
+    setMuralsInCollection((oldMurals: any) => [...oldMurals, addedMural]);
+  };
+
+  const handleSave = () => {
+    if (!title.length || !description.length) return;
+
+    axios.post('http://localhost:3000/collection', {
```
Comment (JDziewonski98) [inline frontend/src/CollectionForm/CollectionForm.tsx:67]: URLs should be parametrized or in constants

File: frontend/src/CollectionForm/CollectionForm.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            onMouseLeave={() => setHoveringDesc(false)}
+            endAdornment={!editingDesc && (
+              <InputAdornment position="start">
+                <EditIcon color={hoveringDesc ? "primary" : "action"} />
+              </InputAdornment>
+            )}
+          />
+          <div className={styles.field}>
+            <Autocomplete
+              freeSolo
```
Comment (JDziewonski98) [inline frontend/src/CollectionForm/CollectionForm.tsx:140]: freeSolo lets a user input anything, correct? Shouldn't it be restricted to only the autocomplete options?

**Comment by JDziewonski98** [comment]
Some nitpicks but otherwise LGTM

File: frontend/src/CollectionForm/CollectionForm.tsx (hunk 3 above)
Comment (tedspare) [reply frontend/src/CollectionForm/CollectionForm.tsx:67]: I'm going to leave these for now for issue #101

**Review by JDziewonski98 — APPROVED** [review]
lgtm

**Review by zhenglinxiao — APPROVED** [review]
LGTM :)

**Pull request merged**